homeassistant/components/viam/config_flow.py

```diff
@@ -0,0 +1,191 @@
+"""Config flow for viam integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from viam.app.app_client import AppClient
+from viam.app.viam_client import ViamClient
+from viam.rpc.dial import Credentials, DialOptions
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_ADDRESS, CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+
+from .const import (
+    CONF_API_ID,
+    CONF_CREDENTIAL_TYPE,
+    CONF_ROBOT,
+    CONF_ROBOT_ID,
+    CONF_SECRET,
+    CRED_TYPE_API_KEY,
+    CRED_TYPE_LOCATION_SECRET,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_AUTH_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CREDENTIAL_TYPE): SelectSelector(
+            SelectSelectorConfig(
+                options=[
+                    SelectOptionDict(value=CRED_TYPE_API_KEY, label="Org API Key"),
+                    SelectOptionDict(
+                        value=CRED_TYPE_LOCATION_SECRET, label="Robot Location Secret"
+                    ),
```

Comment:
The option labels should be in `strings.json`, not here, unless this should not be translated.

Reply:
These are in `strings.json`, which is why the `translation_key` is set: https://github.com/HipsterBrown/home-assistant-core/blob/3103d540017b9dfca1b364e18179306a83febd44/homeassistant/components/viam/strings.json#L40

I thought the `options` still need to be set even if the `translation_key` is set.

Reply:
You of course need to set the options, but you don't need to provide a `SelectOptionDict` with key + label

Reply:
Yes, the harm is that you now have hardcoded English in the Python code + translatable English in `strings.json`. This makes intentions unclear. Please don't use a `SelectOptionDict` here, just use a list with keys.

Reply:
Ok, thanks for providing that feedback. I understand your reasoning now. I've updated the config to just use the `CRED_TYPE_LOCATION_SECRET` and `CRED_TYPE_API_KEY` constant keys.